problems/4.arrays.js

```diff
@@ -1,31 +1,60 @@
 /** Averages the first and last value of an array.  Return null if the array has fewer than 2 elements.
-* @param {number[]} arr - The input array
-* @returns {number} - The average of the first and last values
-*/
-function firstAndLastAverage() {
-
-}
+ * @param {number[]} arr - The input array
+ * @returns {number} - The average of the first and last values
+ */
+//function firstAndLastAverage() {}
+const firstAndLastAverage = (arr) => {
+  let first = arr[0];
+  let last = arr[arr.length - 1];
+  let average = 0;
+  if (arr.length >= 2) {
+    average = (first + last) / 2;
+    return average;
+  } else {
+    return null;
+  }
+};
 
 /** Duplicates all values inside an array.
-* Sample input: [1,4,2,5]
-* Sample output: [1,4,2,5,1,4,2,5]
-* @param {number[]} arr - The input array
-* @returns {number[]} - The duplicated array
-*/
-function duplicateArray() {
-
-}
+ * Sample input: [1,4,2,5]
+ * Sample output: [1,4,2,5,1,4,2,5]
+ * @param {number[]} arr - The input array
+ * @returns {number[]} - The duplicated array
+ */
+//function duplicateArray() {}
+const duplicateArray = (arr) => {
+  // let newArr = arr;
+  // for(i = 0; i < arr.length; i++) {
+  //   newArr.push(arr[i]);
```

Comment:
because newArr and arr both point to the same spot in memory, as you push they both increase in length.

Reply:
I get it now.